mux.go

```diff
@@ -378,11 +378,11 @@ func (mx *Mux) NotFoundHandler() http.HandlerFunc {
 
 // MethodNotAllowedHandler returns the default Mux 405 responder whenever
 // a method cannot be resolved for a route.
-func (mx *Mux) MethodNotAllowedHandler() http.HandlerFunc {
+func (mx *Mux) MethodNotAllowedHandler(methodsAllowed ...methodTyp) http.HandlerFunc {
 	if mx.methodNotAllowedHandler != nil {
 		return mx.methodNotAllowedHandler
 	}
-	return methodNotAllowedHandler
+	return methodNotAllowedHandler(methodsAllowed...)
```

Comment:
Unfortunately I cannot pass the Allowed Methods to the `(Mux).methodNotAllowedHandler` given by the user directly in the Mux with the `MethodNotAllowed()` method, as it would break the current API.

It only set the `Allow` header when using default handler. But I think it is enough anyway, at least for this PR.

We might want to give the possibility for users to implement more easily this feature in their own handler by making the `(Context).methodsAllowed` field public. Is it worth it? It's up to you.